pkg/kms/localkms/localkms_reader.go

```diff
@@ -0,0 +1,68 @@
+/*
+ Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package localkms
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// newReader will create a new local storage storeReader of a keyset with ID value = keysetID
+// it is used internally by local kms
+func newReader(store storage.Store, keysetID string) io.Reader {
+	return &storeReader{
+		storage:  store,
+		keysetID: keysetID,
+	}
+}
+
+// storeReader struct to load a keyset from a local storage
+type storeReader struct {
+	buf      []byte
+	lastRead int
+	storage  storage.Store
+	keysetID string
+}
+
+// Read a marshaled keyset from local storage representing a keyset with ID = l.keysetID and copy it to p
+// the length of the content copied will be len(p)
+func (l *storeReader) Read(p []byte) (int, error) {
+	if l.keysetID == "" {
+		return 0, fmt.Errorf("keysetID is not set")
+	}
+
+	// populate buf if empty
+	if len(l.buf) == 0 {
+		data, err := l.storage.Get(l.keysetID)
+		if err != nil {
+			return 0, err
+		}
+
+		capLengthFactor := 2
+		l.buf = make([]byte, len(data), capLengthFactor*len(data))
```

Comment:
I think with `l.buf` you can now just create and hold on to a `byte.Buffer` and delegate `Read()` calls to it - avoiding the rest of the code below.

Reply:
The function cannot control the length of `p`, it can be either less or more than the content retrieved from the store. The code below is needed if it's smaller than the store's keyset

Reply:
will replace logic with bytes.Buffer

Reply:
done